src/services/attendee/attendee-router.test.ts

```diff
@@ -1,82 +1,239 @@
+
+    await SupabaseDB.EVENTS.insert({
+        eventId: testEventId,
+        name: "Test Event",
+        description: "Description",
+        startTime: new Date().toISOString(),
+        endTime: new Date(Date.now() + 3600000).toISOString(), // +1hr
+        eventType: "SPECIAL", // use one of the allowed enums
+        isVirtual: false,
+        isVisible: true,
+        location: "Test Location",
+        points: 10,
+    }).throwOnError();
+
+    return testEventId;
+}
+
+type RegistrationOverride = {
+    userId?: string;
+    displayName?: string;
+    email?: string;
+    name?: string;
+    university?: string;
+    degree?: string;
+    isInterestedMechMania?: boolean;
+    isInterestedPuzzleBang?: boolean;
+    dietaryRestrictions?: string[];
+    allergies?: string[];
+};
+
+type AttendeeOverride = {
+    userId?: string;
+    displayName?: string;
+    points?: number;
+    favoriteEvents?: string[];
+    isEligibleTshirt?: boolean;
+    isEligibleCap?: boolean;
+    isEligibleTote?: boolean;
+    isEligibleButton?: boolean;
+    hasRedeemedTshirt?: boolean;
+    hasRedeemedCap?: boolean;
+    hasRedeemedTote?: boolean;
+    hasRedeemedButton?: boolean;
+    hasPriorityMon?: boolean;
+    hasPriorityTue?: boolean;
+    hasPriorityWed?: boolean;
+    hasPriorityThu?: boolean;
+    hasPriorityFri?: boolean;
+    hasPrioritySat?: boolean;
+    hasPrioritySun?: boolean;
+    puzzlesCompleted?: string[];
+};
+
+export async function insertTestAttendee(
+    overrides: {
+        registration?: RegistrationOverride;
+        attendee?: AttendeeOverride;
+    } = {}
+) {
+    const userId = TESTER.userId;
+    const email = TESTER.email;
+
+    // Insert role
+    await SupabaseDB.ROLES.insert({
+        userId,
+        displayName: "Test User",
+        email,
+        roles: [Role.enum.USER],
+    }).throwOnError();
+
+    // Insert registration
+    await SupabaseDB.REGISTRATIONS.insert({
+        userId,
+        email,
+        name: "Test User",
+        university: "UIUC", // default test value
+        degree: "BS", // default test value
+        isInterestedMechMania: true,
+        isInterestedPuzzleBang: false,
+        dietaryRestrictions: [],
+        allergies: [],
+        ...overrides.registration,
+    }).throwOnError();
+
+    // Insert attendee
+    await SupabaseDB.ATTENDEES.insert({
+        userId,
+        points: 0,
+        favoriteEvents: [],
+        isEligibleTshirt: true,
+        isEligibleCap: false,
+        isEligibleTote: false,
+        isEligibleButton: false,
+        hasRedeemedTshirt: false,
+        hasRedeemedCap: false,
+        hasRedeemedTote: false,
+        hasRedeemedButton: false,
+        hasPriorityMon: false,
+        hasPriorityTue: false,
+        hasPriorityWed: false,
+        hasPriorityThu: false,
+        hasPriorityFri: false,
+        hasPrioritySat: false,
+        hasPrioritySun: false,
+        puzzlesCompleted: [],
+        ...overrides.attendee,
```

Comment:
yeah just replace this with the database schema imo, or even a shared attendee type from the overall /router or /schema or /validators file, not worth creating your own types in the test file